server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -893,7 +892,7 @@ private IndexShardState changeState(IndexShardState newState, String reason) {
     private Engine.Delete prepareDelete(String id, Term uid, long seqNo, long primaryTerm, long version,
                                                VersionType versionType, Engine.Operation.Origin origin,
                                                long ifSeqNo, long ifPrimaryTerm) {
-        long startTime = System.nanoTime();
+        long startTime = threadPool.relativeTimeInNanos();
```

Comment:
This will conflict with `InternalEngine`'s use of this value in e.g. where we still use the un-cached time:

```java
            deleteResult.setTook(System.nanoTime() - delete.startTime());
```
which will still throw of the `InternalIndexingStats` I think?